src/group_operations/addition.jl

```diff
@@ -0,0 +1,206 @@
+"""
+    AdditionGroupOperation <: AbstractGroupOperation
+
+A group operation that is realised introducing defaults that fall back
+to `+` and `-` being overloaded, for example
+`_compose(G::LieGroup{𝔽,AdditionGroupOperation}, a,b) = a+b`
+"""
+struct AdditionGroupOperation <: AbstractGroupOperation end
+
+#
+#
+# Handle interactions of `+` and `-` with the identity element, though they are
+# also already handled on the `compose()` level
+Base.:+(e::Identity{AdditionGroupOperation}) = e
+Base.:+(e::Identity{AdditionGroupOperation}, ::Identity{AdditionGroupOperation}) = e
+Base.:+(::Identity{AdditionGroupOperation}, g) = g
+Base.:+(g, ::Identity{AdditionGroupOperation}) = g
+
+Base.:-(e::Identity{AdditionGroupOperation}) = e
+Base.:-(e::Identity{AdditionGroupOperation}, ::Identity{AdditionGroupOperation}) = e
+Base.:-(::Identity{AdditionGroupOperation}, g) = -g
+Base.:-(g, ::Identity{AdditionGroupOperation}) = g
+
+_doc_compose_add = """
+    compose(G::LieGroup{𝔽,AdditionGroupOperation}, g, h)
+    compose!(G::LieGroup{𝔽,AdditionGroupOperation}, k, g, h)
+
+Copmute the group operation composition of `g` and `h` with respect to
+the [`AdditionGroupOperation`](@ref) on `G`, which falls back to calling
+`g+h`, where `+` is assumed to be overloaded accordingly.
+
+This can be computed in-place of `k`.
+"""
+
+@doc "$(_doc_compose_add)"
+compose(::LieGroup{𝔽,AdditionGroupOperation}, g, h) where {𝔽}
+
+@doc "$(_doc_compose_add)"
+compose!(::LieGroup{𝔽,AdditionGroupOperation}, k, g, h) where {𝔽}
+
+function _compose!(G::LieGroup{𝔽,AdditionGroupOperation}, k, g, h) where {𝔽}
+    ManifoldsBase.copyto!(G, k, g + h)
```

Comment:
```suggestion
    k .+= g .+ h
```
This should be sufficiently generic to use here.

Reply:
I think this does not work as you think. For the two test `g1, g2 = [1.0, 0.0, 0.0], [0.0, 3.0, 0.0]` sure they sum has a zero third entry, but calling `compose` on them now yields

```
compose(G, g1, g2)
3-element Vector{Float64}:
 1.0
 3.0
 5.2936122611302e-310
```

Ah! The first plus is wrong!